src/Tables.jl

```diff
@@ -1,92 +1,396 @@
 module Tables
 
 using Unidecode
-using GeoParams: AbstractMaterialParam, param_info
+using GeoParams: AbstractMaterialParam, param_info, LinearViscous, PowerlawViscous, DislocationCreep, DiffusionCreep, CompositeRheology, Parallel
 using ..Units
 using ..MaterialParameters: MaterialParamsInfo
 
-export Phase2Dict, Dict2LatexTable
+export detachFloatfromExponent, Phase2Dict, Dict2LatexTable, Phase2DictMd, Dict2MarkdownTable, ParameterTable
 
-# was tested with:
-# TestPh = (SetMaterialParams(Name="Viscous Matrix", Phase=1, Density=ConstantDensity(),CreepLaws = SetDislocationCreep("Quartz Diorite | Hansen & Carter (1982)")),
-#           SetMaterialParams(Name="Viscous Sinker", Phase=2, Density= PT_Density(),CreepLaws = LinearViscous(η=1e21Pa*s)),
-#           SetMaterialParams(Name="Viscous Bottom", Phase=3, Density= PT_Density(),CreepLaws = SetDislocationCreep("Diabase | Caristan (1982)")));
 
-"""
-Phase2Dict() puts all parameters of a phase in a dict.
-Dict2LatexTable() writes .tex file with all parameters from Phase2Dict() output in a table.
+""" 
+detachFloatfromExponent() returns the number of float decimals after the comma as Integer, the float number without the exponent as string 
+and the exponent after the "e" as string.
+The argument output returns "1" for "ex" if the input number has no exponent.
 
 """
+function detachFloatfromExponent(str::String)
+    s = lowercase(str)
+    if 'e' in s
+        s, ex = split(s, "e")
+    else
+        ex = "1"
+    end
 
+    dig = something(findfirst('.', reverse(s)), 1) - 1
+    
+    return dig, s, ex
+
+end
+
+
+"""
+Phase2Dict() puts all parameters of a phase s in a dict.
+"""
 function Phase2Dict(s)
-    # Dict has Key with Fieldname and Value with Tuple(value, symbol, unit)
-    fds = Dict{String,Tuple{String,String,String,String}}()
+    # Dict has Key with Fieldname and Value with Tuple(value, symbol, flowlaw keyword, number of phases, (Disl+Diff) or Lin, Disl or Diff)
+    fds = Dict{String,Tuple{String,String,String,String,String,String}}()
     refs = Dict{String,Tuple{String,String,String}}()
-    # Descriptions for every parameter that could occur in the table and their corresponding variable name(s) that is used in GeoParams
+    if !(typeof(s) <: Tuple)
```

Comment:
This could be a function with a couple of methods in `Utils.jl`, may become handy eventually. Something like:
```julia
make_tuple(x) = (x,)
make_tuple(x::Tuple) = x
```

Reply:
Not quite sure what the second variant of your given example should do, when x is already a tuple .
Also, isn't it easier to just use `(x,)` by its own since `make_tuple` would do the same but with more typing?

Reply:
If you define those 2 methods, you can basically remove the branching. If `x` is already a tuple, then `make_tuple` does nothing and returns `x` itself, whereas if it's not a tuple, it makes it one.  The whole point is that in Julia, you can avoid introducing branching that checks on types by making use of multiple dispatch. And in the end I would say doing `make_tuple(s)` is less verbose and more readable than checking for its type.

Reply:
Ah, ok. I implemented it.